PowerShell/ScubaGear/Modules/CreateReport/CreateReport.psm1

```diff
@@ -168,6 +190,98 @@ function New-Report {
     $ReportSummary
 }
 
+function Import-SecureBaseline{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({Test-Path -PathType Container $_})]
+        [string]
+        $BaselinePath = (Join-Path -Path $PSScriptRoot -ChildPath "..\..\..\..\baselines\")
+    )
+
+    $ProductNames = Get-ChildItem $BaselinePath -Filter "*.md" | ForEach-Object {$_.Name.SubString(0, $_.Name.Length - 3)}
+    $Output = @{}
+
+    foreach ($Product in $ProductNames) {
+        $Output[$Product] = @()
+        $ProductPath = Join-Path -Path $BaselinePath -ChildPath "$Product.md"
+        $MdLines = Get-Content -Path $ProductPath
+
+        # Select-String line numbers aren't 0-indexed, hence the "-1" on the next line
+        $LineNumbers = Select-String "^## [0-9]+\." "$($BaselinePath)$($Product).md" | ForEach-Object {$_."LineNumber"-1}
+        $Groups = $LineNumbers | ForEach-Object {$MdLines[$_]}
+
+        foreach ($GroupName in $Groups) {
+            $Group = @{}
+            $Group.GroupNumber = $GroupName.Split(".")[0].SubString(3) # 3 to remove the "## "
+            $Group.GroupName = $GroupName.Split(".")[1].Trim() # 1 to remove the leading space
+            $Group.Controls = @()
+
+            $IdRegex =  "#### MS\.[$($Product.ToUpper())]+\.$($Group.GroupNumber)\."
+            # Select-String line numbers aren't 0-indexed, hence the "-1" on the next line
+            $LineNumbers = Select-String $IdRegex "$($BaselinePath)$($Product).md" | ForEach-Object {$_."LineNumber"-1}
+
+            foreach ($LineNumber in $LineNumbers) {
+                # This assumes that the value is on the immediate next line after the ID
+                $LineAdvance = 1;
+                $Value = ([string]$MdLines[$LineNumber+$LineAdvance]).Trim()
+
+                while ($Value.Substring($Value.Length-1,1) -ne "."){
+                    $LineAdvance++
+                    $Value += ' ' + ([string]$MdLines[$LineNumber+$LineAdvance]).Trim()
+                }
+
+                $null = $Control.Value -Match '^(.+)(SHALL|SHOULD|MAY){1,1}(.+\.)$'
+                $Criticality = "UNK"
+                if (@("SHALL", "SHOULD", "MAY") -Contains $Matches.2){
+                    $Criticality = $Matches.2
+                }
+
+                $Value = [System.Net.WebUtility]::HtmlEncode($Value)
+                $Id = [string]$MdLines[$LineNumber].Substring(5)
+
+                if ($Id.EndsWith("X")){
+                    $Deleted = $true
+                    $Id = $Id -Replace ".$"
+                    $Value = "[DELETED] " + $Value
+                }
+                else {
+                    $Deleted = $false
+                }
+
+                $Group.Controls += @{"Id"=$Id; "Value"=$Value; "Criticality"=$Criticality; "Deleted"=$Deleted}
```

Comment:
The criticality is determined here but as far as I can tell it is never actually used. Should a TODO item be documented here?

Reply:
Removed parsing of description for 'clues' to criticality.  Opened issue #341